src/main/kotlin/io/getunleash/metrics/MetricsReporter.kt

```diff
@@ -109,32 +109,67 @@ class HttpMetricsReporter(
 
     }
 
-    override fun log(featureName: String, enabled: Boolean): Boolean {
-        val count = if (enabled) {
-            EvaluationCount(1, 0)
-        } else {
-            EvaluationCount(0, 1)
-        }
-        bucket.toggles.merge(featureName, count) { old: EvaluationCount?, new: EvaluationCount ->
-            old?.copy(yes = old.yes + new.yes, no = old.no + new.no) ?: new
+       override fun log(featureName: String, enabled: Boolean): Boolean {
+        try {
+            val count = if (enabled) {
+                EvaluationCount(1, 0)
+            } else {
+                EvaluationCount(0, 1)
+            }
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+                bucket.toggles.merge(
+                    featureName,
+                    count
+                ) { old: EvaluationCount?, new: EvaluationCount ->
+                    old?.copy(yes = old.yes + new.yes, no = old.no + new.no) ?: new
+                }
+            } else {
+                // handle for android 5,6
+                try {
+                    bucket.toggles.remove(featureName)
+                    bucket.toggles[featureName] = count
```

Comment:
this will basically disable metrics for android 5 and 6. You're removing the current state blindly without preserving the previous count.